src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,502 @@
+        corresponding to the referenced dataset config, the associated connector instance (different from the
+        DynamicErasureEmailConnector), and the collection address and fields of the dataset where we need to
+        lookup the recipient email address and vendor name.
+
+        Raises a DynamicErasureEmailConnectorException if the connector configuration is invalid.
+        """
+        # Import here to avoid circular import error
+        # TODO: once we incorporate custom request fields into the DSR graph logic  we won't need this import here,
+        # so ignorning cyclic import warning for now
+        from fides.api.service.connectors import (  # pylint: disable=cyclic-import
+            get_connector,
+        )
+
+        email_dataset_reference = self.config.recipient_email_address
+        email_dataset_key = email_dataset_reference.dataset  # pylint: disable=no-member
+
+        vendor_dataset_reference = self.config.third_party_vendor_name
+        vendor_datset_key = (
+            vendor_dataset_reference.dataset  # pylint: disable=no-member
+        )
+
+        if email_dataset_key != vendor_datset_key:
+            error_log = f"Dynamic Erasure Email Connector with key {self.configuration.key} references different datasets for email and vendor fields. Skipping erasure email send."
+            logger.error(error_log)
+            self.error_all_privacy_requests(
+                db,
+                privacy_requests,
+                "Connector configuration references different datasets for email and vendor fields.",
+            )
+            raise DynamicErasureEmailConnectorException(error_log)
+
+        collection_name, email_field_name = (
+            self.get_collection_and_field_from_reference(
+                db, privacy_requests, self.config.recipient_email_address
+            )
+        )
+
+        vendor_collection_name, vendor_field_name = (
+            self.get_collection_and_field_from_reference(
+                db, privacy_requests, self.config.third_party_vendor_name
+            )
+        )
+
+        if collection_name != vendor_collection_name:
+            error_log = f"Dynamic Erasure Email Connector with key {self.configuration.key} references different collections for email and vendor fields. Skipping erasure email send."
+            logger.error(error_log)
+            self.error_all_privacy_requests(
+                db,
+                privacy_requests,
+                "Connector configuration references different collections for email and vendor fields.",
+            )
+            raise DynamicErasureEmailConnectorException(error_log)
+
+        # We get the DatasetConfig instance using the dataset key provided in the configuration
+        dataset_config: Optional[DatasetConfig] = DatasetConfig.get_by(
+            db, field="fides_key", value=email_dataset_key
+        )
+
+        if not dataset_config:
```

Comment:
Didn't we talk about the possibility of multiple matches and raising an error there as well?

Reply:
ah yes but that turned out not to be the case, I was just very confused because the fides_key for the connection configs usually match those in the dataset. I double and triple checked, also validated with Adrian, that the dataset reference field points to a single DatasetConfig, not the Dataset itself. This is assuming the fides_key of a DatasetConfig is unique , otherwise we could still have the issue I think

Reply:
OK thanks for clarifying this was confusing to me too